CAIPs/caip-18.md

```diff
@@ -0,0 +1,67 @@
+---
+caip: 18
+title: NFT ID Specification
+author: Joel Thorstensson (@oed)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/18
+status: Draft
+type: Standard
+created: 2020-04-23
+updated: 2020-04-23
+requires: 10
+---
+
+## Simple Summary
+
+CAIP-18 defines a way to identify a specific NFT token on any blockchain specified by CAIP-2 blockchain id.
+
+## Abstract
+
+This proposal aims to facilitate specifying NFT tokens on any blockchain extending CAIP-2 blockchain id specification. This is useful for both decentralized applications and wallets to specify NFT tokens on multiple chains using string identifiers specific to each chain. Currently wallets are usually designed for each chain and multi-chain wallets use proprietary data structures to differentiate NFTs. This proposal aims to standardize these identifiers for NFTs to allow interoperability.
+
+## Motivation
+
+The motivation for proposal stem from designing a chain-agnostic protocol for communication between dapps and wallets that was independent of any blockchain but provide the flexibility to be backwards compatible with existing applications.
+
+## Specification
+
+The NFT id specification will be prefixed with the NFT token ID and the NFT contract address and delimited with an at sign (`@`) followed by a CAIP-2 blockchain ID. Since [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) can be used to reference a specific account (which includes contract addresses) we can represent the NFT id as the token specific id concatinated with the CAIP-10 account id.
```

Comment:
For a Cosmos NFT, I think there is no NFT contract address ( because no smart contract, but only natives modules https://github.com/cosmos/modules/tree/master/incubator/nft and just transactions types https://github.com/cosmos/modules/blob/master/incubator/nft/docs/spec/03_messages.md).

Reply:
Thanks for providing insight @ntoineherzog! So what do you need to uniquely identify an NFT on cosmos? Just the ID? Or do you also need the Denom?

Reply:
One more thing, this spec is still draft I think and I believe things will get clarifies after Games of Zones when we actually have NFT moving from chain A to chain B. To give some context, at IOV, we don't implement yet the NFT spec because we are waiting for to see how things will mature. So this spec is very early stage.

I think you are mixing the concept of address and smart contract. In the ethereum world, an address can be smart contract so it makes sense to mix the concept in your current CAIP for ethereum but it makes no sense for native chains like cosmos.

Also I would propose something  more like an URL because you can think it like a path

So you should start with the CAIP2
so something like with the chain_ID
eip155:1/type_of_asset

type_of_asset could be ERC2O token or NFT, for ethereum it is going to be a smart contract address but for cosmos it is going to be the TICKER or the name of the NFT collection

then for NFT, you can add later the ID of the NFT so something like that at the end:

eip155:1/type_of_asset/ID_of the NFT

example for DAI

eip155:1/0x6b175474e89094c44da98b954eedeac495271d0f

example for ATOM

cosmos:comoshub-3/ATOM

example for crypto-kitties NFT with ID 1

eip155:1/0x06012c8cf97BEaD5deAe237070F9587f8E7A266d/1

I hope it helps!

Reply:
what do you think @pedrouid ?

Reply:
i have no preference for the order of ids but agree with @antoineherzog that `denom` should be the stand in for contract address on cosmos-like chains. `denom` does not change per asset, although there is currently no enforced character or size limit on the string which makes up denom. This is the best draft of an Interchain Standard for NFT transfers to date: https://github.com/cosmos/ics/pull/294/files. It seems to follow the spec implemented for fungible token transfers here: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer. Neither seem to have restrictions on denom strings.